app/url_utils.py

```diff
@@ -0,0 +1,24 @@
+from urllib.parse import urlsplit, urlunsplit
```

Comment:
I'd drop this in the multi purpose utils file. Seems overkill to have another file for a single function.

Reply:
Personally I think multi-purpose utils files are the spawn of the devil. They only ever get longer and longer and longer and before you know it you have a 4000 line pile of spew. I've seen it (in a previous employment that had existed for longer than GDS, admittedly).

Reply:
Unless we're planning on splitting out that other file into different files, ie `json_utils`, `request_utils`, `data_utils` then an extra file is inconsistent. :man_shrugging:

Reply:
Got to start somewhere.

Reply:
Quite: councils of perfection that result in complete inaction aren't the incremental agile change we're signed up to ;-) In this context, inconsistency is inevitable and good, as long as we're agreed about the direction of travel.

Reply:
Incidentally I wanted to put this into dmutils, but I won't be doing that cos supporting python 2 is a PITA :-p

Reply:
Creating inconsistencies in the code base can make you feel like you're doing the _right_ thing, and you might well be doing the right thing, in this case you probably are, butmy problem is that it inconsistency _reduces confidence_ in the code. People will be confused by the multiple conventions and will wonder why that code had to be kept separately, or be tacked on, or not be properly integrated.

It doesn't reduce confidence in all the other code that follows the same patterns (no matter how wrong they might be); It reduces confidence in the different code, the inconsistent code.

I personally thing that we should be using class based views for all our views but if I started writing class based views it would look weird and out of place. And besides, that wouldn't achieve my goal of all views being class based. It would just mean we had 2 paradigms until I classed up all the views.

It's all about the easiest way for you to achieve your goal of not having a big utils file:
Say that it falls outside the scope of this ticket. (If this conversation was the only thing holding up this PR how annoying would that be?)
Highlight it to other members of the team to make sure they agree.
Then either:
 * Create a tech debt ticket and get someone else to do it in a separate PR all at once 🎉 
 * Do it yourself in a separate PR all at once
That way the PR will be accepted quickly because people will see it's just arefacotr and not have to go digging through a refactor for functional changes

Half staring a refactor and tagging it onto a bug fix PR doesn't help you achieve your goal of not having a massive utils file. It makes the PR harder to read, introduces confusion and reduces confidence.

I'm not trying to 💩  on the idea or anything like that. I think breaking the code into other files is a good idea. But I think that if we do it, we should do the whole thing. Every code base I've ever worked with has had a legacy of accepting PRs that introduce new styles and paradigms and conventions without evaluating closely enough the effort it will take to implement them fully. In this case the effort is low, but the cost of failing to complete the refactor is high (reduced confidence)

That's why I think consistency is important, even if what you're being consistent with isn't quite perfect.

That's the last I'll say on it, it's up to you if you want to change it.